src/firebolt_ingest/table_service.py

```diff
@@ -341,3 +341,41 @@ def does_internal_table_exist(self) -> bool:
         Checks if the internal table exists in the database.
         """
         return does_table_exist(self.connection.cursor(), self.internal_table_name)
+
+    def drop_outdated_partitions(self):
+        """
+        Drops partitions in the fact table that are outdated, meaning the corresponding
+            file in the external table has a more recent timestamp (was updated).
+        """
+        cursor = self.connection.cursor()
+        if not does_table_exist(cursor, self.internal_table_name):
+            raise FireboltError(f"Fact table {self.internal_table_name} doesn't exist")
+        if not does_table_exist(cursor, self.external_table_name):
+            raise FireboltError(
+                f"External table {self.external_table_name} doesn't exist"
+            )
+        if not self.table.partitions:
+            raise FireboltError(
+                f"Fact table {self.external_table_name} is not partitioned"
+            )
+
+        outdated_partitions_query = f"SELECT DISTINCT {','.join([p.as_sql_string() for p in self.table.partitions])} \
+        FROM {self.external_table_name} \
+        WHERE source_file_timestamp > ( SELECT MAX(source_file_timestamp) \
+            FROM {self.internal_table_name} )"
+
+        cursor.execute(query=format_query(outdated_partitions_query))
+        outdated_partitions = cursor.fetchall()
+        logger.info(f"List of outdated partitions: {outdated_partitions}")
+
+        if outdated_partitions:
+            q = "ALTER TABLE {table_name} DROP PARTITION {partition_expression}"
+            for outdated_partition in outdated_partitions:
+                logger.info(
+                    f"Going to drop the following partitions: {outdated_partition}"
+                )
```

Comment:
Maybe let's make this a debug level, WDYT? I'm just afraid it might be clogging up the log if there are many partitions.

Reply:
Done